test/dynamo/test_modes.py

```diff
@@ -660,6 +660,43 @@ def forward(self, x):
         with torch.device("cpu"):
             torch.compile(mod, fullgraph=True)(x)
 
+    @requires_gpu
+    def test_hop(self):
+        import torch
+        import torch._higher_order_ops
+        from torch.nn.attention.flex_attention import flex_attention
+
+        torch.set_default_device(device_type)
+
+        flex_attention = torch.compile(flex_attention, dynamic=False)
+
+        prefix_lengths = torch.arange(8)
+
+        def prefix_lm(b, h, q, kv):
+            return prefix_lengths[b] >= kv
+
+        class TestMode(BaseTorchFunctionMode):
+            def __torch_function__(self, func, types, args, kwargs=None):
+                if not kwargs:
+                    kwargs = {}
+
+                if func == torch._higher_order_ops.flex_attention:
+                    raise RuntimeError("test")
```

Comment:
nit: it's a bit better to subclass RuntimeError and then check that we are raising that subclass